商品一覧カテゴリー別表示

Ken-Yamamoto opened the pull request:
#WHAT
商品一覧カテゴリー別表示
#WHY
見栄え　および　売り上げにつながるため

Ken-Yamamoto committed:
Show items related with categories on index page

items_controller.rb 追記
category.rb 変更
item_category.rb 変更
/items/にpartial 作成_contentも移動

Ken-Yamamoto commented:
[![Image from Gyazo](https://i.gyazo.com/49ed54a97217338ebd78421dd1273e86.gif)](https://gyazo.com/49ed54a97217338ebd78421dd1273e86)

Ken-Yamamoto committed:
Modify files

Ken-Yamamoto committed:
Modify files

Ken-Yamamoto committed:
Merge branch 'master' into 商品一覧カテゴリー別表示

Ken-Yamamoto committed:
Modify file

ken-nag commented on app/controllers/items_controller.rb:
```diff
@@ -1,7 +1,19 @@
+  INTERIOR = 4
+  BOOKS    = 5
+  HOBYS    = 6
+  COSME    = 7
```
定数を使うときはfreezeしましょう

Ken-Yamamoto replied:
凍結記述完了です。

ken-nag commented on app/models/item_category.rb:
```diff
@@ -1,4 +1,4 @@
-  belongs_to :item
-  belongs_to :category
+  # belongs_to :item
+  # belongs_to :category
```
不要なら削除しましょう

Ken-Yamamoto replied:
削除完了です。

ken-nag commented on app/views/items/_cosme.html.haml:
```diff
@@ -0,0 +1,17 @@
+.main-box-content.clearfix
+  - @items_for_cosme.first(4).each do |item|
```
first(4)の部分はcontrollerで使用するのはどうでしょうか

Ken-Yamamoto replied:
ありがとうございます、controllerへ記載しました！

Ken-Yamamoto committed:
Modify files

items_controller.rb 定数を変更不可にするため凍結処理
item_category.rb 不要な記載を削除
Indexページ.first記述をitems_controller.rbへ移動

TOMARUMARU commented on app/controllers/items_controller.rb:
```diff
@@ -1,7 +1,27 @@
+    @items_for_man = Category.get_category_items(MAN).first(4)
+    @items_for_babykids = Category.get_category_items(BABYKIDS).first(4)
+    @items_for_hobys = Category.get_category_items(HOBYS).first(4)
+    @items_for_cosme = Category.get_category_items(COSME).first(4)
```
同じ処理の繰り返しなので、scopeやメソッドを使いましょう

Ken-Yamamoto replied:
scopeを使用し書き換えました。確認よろしくお願いします。

Ken-Yamamoto replied:
すいません、解消しました。再度お願いします。

Ken-Yamamoto committed:
Modify files to add scope for readability

shingoteshima commented:
Siderのエラーを解消しましょう。

Ken-Yamamoto committed:
Modify files

TOMARUMARU reviewed (approved):
LGTM
